### PR TITLE
Feature - #5708 - Add data to CSV export with additional data

### LINKED ISSUE #5708: Add data to CSV export with additional information
(Former title: A column for the tW type (Key terms, Other terms, Names) should be added to the translationWords_CheckInformation.csv and selections.csv files)

Since we allow distinguishing between them in the app, it makes sense to do the same in the related csv files. It also may be important for those who would want to check manually using the spreadsheet.

## Specs 
- [x] tW export 
  - [x] Add `Type` column on tW export. Fill with `Key terms`, `Other terms`, and `Names`
- [x] tN export
  - [x] Add Priority 1 to all checks
  - [x] Add `Type` column, fill with the `Head Category`
  - [x] Add `GL Quote` column
  - [x] Add `Occurrence Note` column

image

### PR DESCRIPTION
**Pull request by richmahn**
#### Describe what your pull request addresses (Please include relevant issue numbers):
- Fixes #5708

#### Please include detailed Test instructions for your pull request:
- Pick a Titus project, do a bunch of selections, edits, reminders, and comments for all three tools. Do this in both English and Hindi selected as the GL when you go into the tool 
- Go back to project management page and do a CSV Export from the context menu of that project
- Examine that the CheckInformation.csv files for tW and tN both have GL Quotes (previously tW didn't, now they are from the English, as currently there is no way to know what the user's GL is for this)
- For Selections, Reminders, and Comments the GL they had picked in the tool should be shown correctly. Look at all columns and make sure everything is standard, and if information isn't available there is a `N/A` in the column (per Ben's request).
- Also check External Edits, as they shouldn't show GL info (just "N/A") on their rows of the verse edits.
- Word Alignment verse edits also will show "N/A" for GL columns.

Here's a sample export of a Titus project: 
[en_ult_tit_book.zip](https://github.com/unfoldingWord-dev/translationCore/files/3315170/en_ult_tit_book.zip)

#### Standard Test Instructions for PR Review Process:

- [ ] Double check unit tests that have been written
- [ ] Check for documentation for code changes
- [ ] Check that there are not inadvertent commits to tC Apps when reviewing a tC Core PR
- [ ] Checkout the branch locally and ensure that app runs as expected
  - [ ] Ensure tests pass
  - [ ] Open and watch the console for errors
  - [ ] Make sure all actions perform as expected
  - [ ] Import and Load a new Project
  - [ ] Load a tool and perform basic actions
  - [ ] Switch tools and perform basic actions
  - [ ] Switch project to an existing project
  - [ ] Load a tool and perform basic actions
  - [ ] Switch tools and perform basic actions
  - [ ] Next time reverse the order of importing after loading an existing project
- [ ] Reviewer should double check the DoD in the ISSUE, including the “spirit” of the story
- [ ] Ask Ben or Koz if you have any concerns about implementation (especially UI related)

---
This change is [Reviewable](https://reviewable.io/reviews/unfoldingword-dev/translationcore/6144)

### CODE REVIEW

**Comment by dsm-git** [comment]
[Linux build 1.2.0 (d95db77)](https://tc-builds.door43.org/build/feature-richmahn-5708/translationCore-linux-x64-1.2.0-d95db77.deb) is ready for download.
[macOS build 1.2.0 (d95db77)](https://tc-builds.door43.org/build/feature-richmahn-5708/translationCore-macos-x64-1.2.0-d95db77.dmg) is ready for download.
[Windows x64 build 1.2.0 (d95db77)](https://tc-builds.door43.org/build/feature-richmahn-5708/translationCore-win-x64-1.2.0-d95db77.setup.exe) is ready for download.
[Windows x32 build 1.2.0 (d95db77)](https://tc-builds.door43.org/build/feature-richmahn-5708/translationCore-win-x32-1.2.0-d95db77.setup.exe) is ready for download.

**Comment by dsm-git** [comment]
[Linux build 1.2.0 (22e5a1d)](https://tc-builds.door43.org/build/feature-richmahn-5708/translationCore-linux-x64-1.2.0-22e5a1d.deb) is ready for download.
[macOS build 1.2.0 (22e5a1d)](https://tc-builds.door43.org/build/feature-richmahn-5708/translationCore-macos-x64-1.2.0-22e5a1d.dmg) is ready for download.
[Windows x64 build 1.2.0 (22e5a1d)](https://tc-builds.door43.org/build/feature-richmahn-5708/translationCore-win-x64-1.2.0-22e5a1d.setup.exe) is ready for download.
[Windows x32 build 1.2.0 (22e5a1d)](https://tc-builds.door43.org/build/feature-richmahn-5708/translationCore-win-x32-1.2.0-22e5a1d.setup.exe) is ready for download.

**Review by PhotoNomad0 — CHANGES_REQUESTED** [review]
> Reviewed 5 of 555 files at r1, 3 of 13 files at r2.
> *[Reviewable](https://reviewable.io/reviews/unfoldingword-dev/translationcore/6144)* status: 8 of 564 files reviewed, 1 unresolved discussion (waiting on @richmahn)

___
*[src/js/helpers/ProjectAPI.js, line 416 at r2](https://reviewable.io/reviews/unfoldingword-dev/translationcore/6144#-Lha8cDoBkOmhRl6Oe1q:-Lha8cDoBkOmhRl6Oe1r:bdh099n) ([raw file](https://github.com/unfoldingword-dev/translationcore/blob/22e5a1d39dfd5ef364676e9d8096de18bdd034ce/src/js/helpers/ProjectAPI.js#L416)):*
> ```JavaScript
>     Object.keys(parentCategoryMapping).forEach(pcn => {
>       if (parentCategoryMapping[pcn].includes(groupId)) {
>         parentCategoryName = pcn;
> ```

this approach doesn't exit when match is found.  Probably should change for improved performance.  standard JS for loops have fastest performance.

**Comment by dsm-git** [comment]
[Linux build 1.2.0 (4c8a330)](https://tc-builds.door43.org/build/feature-richmahn-5708/translationCore-linux-x64-1.2.0-4c8a330.deb) is ready for download.
[macOS build 1.2.0 (4c8a330)](https://tc-builds.door43.org/build/feature-richmahn-5708/translationCore-macos-x64-1.2.0-4c8a330.dmg) is ready for download.
[Windows x64 build 1.2.0 (4c8a330)](https://tc-builds.door43.org/build/feature-richmahn-5708/translationCore-win-x64-1.2.0-4c8a330.setup.exe) is ready for download.
[Windows x32 build 1.2.0 (4c8a330)](https://tc-builds.door43.org/build/feature-richmahn-5708/translationCore-win-x32-1.2.0-4c8a330.setup.exe) is ready for download.

**Comment by dsm-git** [comment]
[Linux build 1.2.0 (b68a0bc)](https://tc-builds.door43.org/build/feature-richmahn-5708/translationCore-linux-x64-1.2.0-b68a0bc.deb) is ready for download.
[macOS build 1.2.0 (b68a0bc)](https://tc-builds.door43.org/build/feature-richmahn-5708/translationCore-macos-x64-1.2.0-b68a0bc.dmg) is ready for download.
[Windows x64 build 1.2.0 (b68a0bc)](https://tc-builds.door43.org/build/feature-richmahn-5708/translationCore-win-x64-1.2.0-b68a0bc.setup.exe) is ready for download.
[Windows x32 build 1.2.0 (b68a0bc)](https://tc-builds.door43.org/build/feature-richmahn-5708/translationCore-win-x32-1.2.0-b68a0bc.setup.exe) is ready for download.

**Comment by dsm-git** [comment]
[Linux build 1.2.0 (8d4c425)](https://tc-builds.door43.org/build/feature-richmahn-5708/translationCore-linux-x64-1.2.0-8d4c425.deb) is ready for download.
[macOS build 1.2.0 (8d4c425)](https://tc-builds.door43.org/build/feature-richmahn-5708/translationCore-macos-x64-1.2.0-8d4c425.dmg) is ready for download.
[Windows x64 build 1.2.0 (8d4c425)](https://tc-builds.door43.org/build/feature-richmahn-5708/translationCore-win-x64-1.2.0-8d4c425.setup.exe) is ready for download.
[Windows x32 build 1.2.0 (8d4c425)](https://tc-builds.door43.org/build/feature-richmahn-5708/translationCore-win-x32-1.2.0-8d4c425.setup.exe) is ready for download.

**Comment by richmahn** [comment]
@benjore Two questions on this:

1) I made this add the glQuote and occurrenceNotes to ALL CSV files that have quote/book/chapter/verse columns as they were already reusing the same code/function. This means all CSV files will have the new glQuote and occurrenceNote columns even if it only has tW info (which doesn't have those fields in its context. I would think glQuote would be relevant but we don't have that in the contextId like we do for tN)

2) In your example, unlike the other current columns, I didn't name these "GL quote" and "Occurrence note", but kept the same naming convention as "glQuote" and "occurrrenceNote". Is that an issue?

For an example CSV export, please see the CSV files in this zip: 
[en_ult_tit_book.zip](https://github.com/unfoldingWord-dev/translationCore/files/3306818/en_ult_tit_book.zip)

**Review by PhotoNomad0 — CHANGES_REQUESTED** [review]
> Reviewed 3 of 555 files at r1, 1 of 3 files at r3, 2 of 4 files at r4, 3 of 5 files at r5.
> *[Reviewable](https://reviewable.io/reviews/unfoldingword-dev/translationcore/6144)* status: 13 of 525 files reviewed, 1 unresolved discussion (waiting on @richmahn)

___
*[tC_apps/translationWords, line 1 at r5](https://reviewable.io/reviews/unfoldingword-dev/translationcore/6144#-LhkNMPkDQ0Sgp_5o2iz:-LhkNMPkDQ0Sgp_5o2j-:blvh7ak) ([raw file](https://github.com/unfoldingword-dev/translationcore/blob/8d4c425cf5dbb64e43f24d29f43bff0c1fd61b26/tC_apps/translationWords#L1)):*
> ```
> Subproject commit 7213c4d15b5c9eacccfdb73cce6e5a39f87c5d22
> ```

Wrong commit id

**Review by PhotoNomad0 — COMMENTED** [review]
> Reviewed 2 of 555 files at r1.
> *[Reviewable](https://reviewable.io/reviews/unfoldingword-dev/translationcore/6144)* status: 15 of 525 files reviewed, 1 unresolved discussion (waiting on @richmahn)

**Review by PhotoNomad0 — DISMISSED** [review]
> Reviewed 499 of 555 files at r1, 9 of 13 files at r2, 1 of 2 files at r6.
> *[Reviewable](https://reviewable.io/reviews/unfoldingword-dev/translationcore/6144)* status: :shipit: complete! all files reviewed, all discussions resolved

**Comment by dsm-git** [comment]
[Linux build 1.2.0 (257a3a4)](https://tc-builds.door43.org/build/feature-richmahn-5708/translationCore-linux-x64-1.2.0-257a3a4.deb) is ready for download.
[macOS build 1.2.0 (257a3a4)](https://tc-builds.door43.org/build/feature-richmahn-5708/translationCore-macos-x64-1.2.0-257a3a4.dmg) is ready for download.
[Windows x64 build 1.2.0 (257a3a4)](https://tc-builds.door43.org/build/feature-richmahn-5708/translationCore-win-x64-1.2.0-257a3a4.setup.exe) is ready for download.
[Windows x32 build 1.2.0 (257a3a4)](https://tc-builds.door43.org/build/feature-richmahn-5708/translationCore-win-x32-1.2.0-257a3a4.setup.exe) is ready for download.

File: src/js/helpers/csvHelpers.js [hunk 1]
```diff
@@ -6,96 +6,124 @@ import csv from 'csv';
...
+};
+
+/**
+ * Flattens a quote which may be an array of words
+ * @param {*} quote
+ * @returns {string}
+ */
+export const flattenQuote = quote => {
+  if (Array.isArray(quote)) {
+    quote = quote.map(item => item.word).join(" ");
```
Comment (mannycolon) [inline src/js/helpers/csvHelpers.js:127]: How are you handling punctuation ? it looks like you're not

**Review by mannycolon — CHANGES_REQUESTED** [review]
Great work Rich. I left a question/comment about quote arrays with punctuations

**Comment by dsm-git** [comment]
[Linux build 1.2.0 (e52b7d8)](https://tc-builds.door43.org/build/feature-richmahn-5708/translationCore-linux-x64-1.2.0-e52b7d8.deb) is ready for download.
[macOS build 1.2.0 (e52b7d8)](https://tc-builds.door43.org/build/feature-richmahn-5708/translationCore-macos-x64-1.2.0-e52b7d8.dmg) is ready for download.
[Windows x64 build 1.2.0 (e52b7d8)](https://tc-builds.door43.org/build/feature-richmahn-5708/translationCore-win-x64-1.2.0-e52b7d8.setup.exe) is ready for download.
[Windows x32 build 1.2.0 (e52b7d8)](https://tc-builds.door43.org/build/feature-richmahn-5708/translationCore-win-x32-1.2.0-e52b7d8.setup.exe) is ready for download.

File: src/js/helpers/csvHelpers.js (hunk 1 above)
Comment (richmahn) [reply src/js/helpers/csvHelpers.js:127]: Good point. Do we already have a function that does this?

**Comment by dsm-git** [comment]
[Linux build 1.2.0 (34a4831)](https://tc-builds.door43.org/build/feature-richmahn-5708/translationCore-linux-x64-1.2.0-34a4831.deb) is ready for download.
[macOS build 1.2.0 (34a4831)](https://tc-builds.door43.org/build/feature-richmahn-5708/translationCore-macos-x64-1.2.0-34a4831.dmg) is ready for download.
[Windows x64 build 1.2.0 (34a4831)](https://tc-builds.door43.org/build/feature-richmahn-5708/translationCore-win-x64-1.2.0-34a4831.setup.exe) is ready for download.
[Windows x32 build 1.2.0 (34a4831)](https://tc-builds.door43.org/build/feature-richmahn-5708/translationCore-win-x32-1.2.0-34a4831.setup.exe) is ready for download.

**Comment by dsm-git** [comment]
[Linux build 1.2.0 (58536fc)](https://tc-builds.door43.org/build/feature-richmahn-5708/translationCore-linux-x64-1.2.0-58536fc.deb) is ready for download.
[macOS build 1.2.0 (58536fc)](https://tc-builds.door43.org/build/feature-richmahn-5708/translationCore-macos-x64-1.2.0-58536fc.dmg) is ready for download.
[Windows x64 build 1.2.0 (58536fc)](https://tc-builds.door43.org/build/feature-richmahn-5708/translationCore-win-x64-1.2.0-58536fc.setup.exe) is ready for download.
[Windows x32 build 1.2.0 (58536fc)](https://tc-builds.door43.org/build/feature-richmahn-5708/translationCore-win-x32-1.2.0-58536fc.setup.exe) is ready for download.

File: src/js/helpers/csvHelpers.js (hunk 1 above)
Comment (richmahn) [reply src/js/helpers/csvHelpers.js:127]: Fixing

**Comment by dsm-git** [comment]
[Linux build 1.2.0 (bd417b1)](https://tc-builds.door43.org/build/feature-richmahn-5708/translationCore-linux-x64-1.2.0-bd417b1.deb) is ready for download.
[macOS build 1.2.0 (bd417b1)](https://tc-builds.door43.org/build/feature-richmahn-5708/translationCore-macos-x64-1.2.0-bd417b1.dmg) is ready for download.
[Windows x64 build 1.2.0 (bd417b1)](https://tc-builds.door43.org/build/feature-richmahn-5708/translationCore-win-x64-1.2.0-bd417b1.setup.exe) is ready for download.
[Windows x32 build 1.2.0 (bd417b1)](https://tc-builds.door43.org/build/feature-richmahn-5708/translationCore-win-x32-1.2.0-bd417b1.setup.exe) is ready for download.

File: src/js/helpers/csvHelpers.js (hunk 1 above)
Comment (richmahn) [reply src/js/helpers/csvHelpers.js:127]: Fixed

**Comment by dsm-git** [comment]
[Linux build 1.2.0 (d415afc)](https://tc-builds.door43.org/build/feature-richmahn-5708/translationCore-linux-x64-1.2.0-d415afc.deb) is ready for download.
[macOS build 1.2.0 (d415afc)](https://tc-builds.door43.org/build/feature-richmahn-5708/translationCore-macos-x64-1.2.0-d415afc.dmg) is ready for download.
[Windows x64 build 1.2.0 (d415afc)](https://tc-builds.door43.org/build/feature-richmahn-5708/translationCore-win-x64-1.2.0-d415afc.setup.exe) is ready for download.
[Windows x32 build 1.2.0 (d415afc)](https://tc-builds.door43.org/build/feature-richmahn-5708/translationCore-win-x32-1.2.0-d415afc.setup.exe) is ready for download.

**Comment by dsm-git** [comment]
[Linux build 1.2.0 (6a82688)](https://tc-builds.door43.org/build/feature-richmahn-5708/translationCore-linux-x64-1.2.0-6a82688.deb) is ready for download.
[macOS build 1.2.0 (6a82688)](https://tc-builds.door43.org/build/feature-richmahn-5708/translationCore-macos-x64-1.2.0-6a82688.dmg) is ready for download.
[Windows x64 build 1.2.0 (6a82688)](https://tc-builds.door43.org/build/feature-richmahn-5708/translationCore-win-x64-1.2.0-6a82688.setup.exe) is ready for download.
[Windows x32 build 1.2.0 (6a82688)](https://tc-builds.door43.org/build/feature-richmahn-5708/translationCore-win-x32-1.2.0-6a82688.setup.exe) is ready for download.

**Review by PhotoNomad0 — CHANGES_REQUESTED** [review]
> Reviewed 1 of 6 files at r7, 3 of 7 files at r9, 2 of 6 files at r10.
> *[Reviewable](https://reviewable.io/reviews/unfoldingword-dev/translationcore/6144)* status: 521 of 527 files reviewed, 2 unresolved discussions (waiting on @mannycolon, @PhotoNomad0, and @richmahn)

___
*[src/js/common/constants.js, line 10 at r10](https://reviewable.io/reviews/unfoldingword-dev/translationcore/6144#-LiD9TNn0UupRrlaWuqX:-LiD9TNn0UupRrlaWuqY:b28nf1) ([raw file](https://github.com/unfoldingword-dev/translationcore/blob/6a826882ab19a4d59f34b5b0cc4c972d899df0bb/src/js/common/constants.js#L10)):*
> ```JavaScript
> export const PROJECTS_PATH = path.join(ospath.home(), 'translationCore', 'projects');
> export const USER_RESOURCES_PATH = path.join(ospath.home(), "translationCore", "resources");
> export const THELPS_RESOURCES_PATH = path.join(USER_RESOURCES_PATH, 'en', 'translationHelps');
> ```

Should tHelps be hard-coded to English?  I know those are the only helps now, but soon will have Hindi.  Maybe this should be a method?  Or at very least be `THELPS_EN_RESOURCES_PATH`.
___
*[src/js/helpers/csvHelpers.js, line 116 at r6](https://reviewable.io/reviews/unfoldingword-dev/translationcore/6144#-LhCVsnsS1cnc_eltEl2-r6-116:-LiDBSuz9Dt9Em6FrE5H:b-s8jwps) ([raw file](https://github.com/unfoldingword-dev/translationcore/blob/257a3a42fb00c9bbf9b4b4a00e52fdd66cfb2abe/src/js/helpers/csvHelpers.js#L116)):*
<details><summary><i>Previously, richmahn (Richard Mahn) wrote…</i></summary><blockquote>

Fixed
</blockquote></details>

Is this done right?  I thought the idea was to include any punctuation that was in the quote.

**Review by PhotoNomad0 — COMMENTED** [review]
> Reviewed 1 of 6 files at r10.
> *[Reviewable](https://reviewable.io/reviews/unfoldingword-dev/translationcore/6144)* status: 522 of 527 files reviewed, 2 unresolved discussions (waiting on @mannycolon, @PhotoNomad0, and @richmahn)

___
*[src/js/helpers/csvHelpers.js, line 116 at r6](https://reviewable.io/reviews/unfoldingword-dev/translationcore/6144#-LhCVsnsS1cnc_eltEl2-r6-116:-LiDDQFE-AJrWJ80OKuS:bd2z5bv) ([raw file](https://github.com/unfoldingword-dev/translationcore/blob/257a3a42fb00c9bbf9b4b4a00e52fdd66cfb2abe/src/js/helpers/csvHelpers.js#L116)):*
<details><summary><i>Previously, PhotoNomad0 (Bruce McLean) wrote…</i></summary><blockquote>

Is this done right?  I thought the idea was to include any punctuation that was in the quote.
</blockquote></details>

Manny, is this correct?

Are we going to have to add other punctuation as we come across it?  I can think of left-hand and right-hand double quotes.

**Review by PhotoNomad0 — COMMENTED** [review]
> Reviewed 2 of 7 files at r9, 3 of 6 files at r10.
> *[Reviewable](https://reviewable.io/reviews/unfoldingword-dev/translationcore/6144)* status: all files reviewed, 2 unresolved discussions (waiting on @mannycolon and @richmahn)

**Review by richmahn — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/unfoldingword-dev/translationcore/6144)* status: all files reviewed, 2 unresolved discussions (waiting on @mannycolon and @PhotoNomad0)

___
*[src/js/common/constants.js, line 10 at r10](https://reviewable.io/reviews/unfoldingword-dev/translationcore/6144#-LiD9TNn0UupRrlaWuqX:-LiE8ffX4Ug3smUIddDr:b-we8vcv) ([raw file](https://github.com/unfoldingword-dev/translationcore/blob/6a826882ab19a4d59f34b5b0cc4c972d899df0bb/src/js/common/constants.js#L10)):*
<details><summary><i>Previously, PhotoNomad0 (Bruce McLean) wrote…</i></summary><blockquote>

Should tHelps be hard-coded to English?  I know those are the only helps now, but soon will have Hindi.  Maybe this should be a method?  Or at very least be `THELPS_EN_RESOURCES_PATH`.
</blockquote></details>

Done. Renamed.

**Review by PhotoNomad0 — APPROVED** [review]
> Reviewed 2 of 2 files at r11.
> *[Reviewable](https://reviewable.io/reviews/unfoldingword-dev/translationcore/6144)* status: all files reviewed, 1 unresolved discussion (waiting on @mannycolon)

**Comment by PhotoNomad0** [comment]
CODE - LGTM waiting on build

**Comment by dsm-git** [comment]
[Linux build 1.2.0 (6f83645)](https://tc-builds.door43.org/build/feature-richmahn-5708/translationCore-linux-x64-1.2.0-6f83645.deb) is ready for download.
[macOS build 1.2.0 (6f83645)](https://tc-builds.door43.org/build/feature-richmahn-5708/translationCore-macos-x64-1.2.0-6f83645.dmg) is ready for download.
[Windows x64 build 1.2.0 (6f83645)](https://tc-builds.door43.org/build/feature-richmahn-5708/translationCore-win-x64-1.2.0-6f83645.setup.exe) is ready for download.
[Windows x32 build 1.2.0 (6f83645)](https://tc-builds.door43.org/build/feature-richmahn-5708/translationCore-win-x32-1.2.0-6f83645.setup.exe) is ready for download.

**Pull request merged**